src/libraries/System.Diagnostics.DiagnosticSource/tests/ActivityTests.cs

```diff
@@ -1694,6 +1694,21 @@ public void StructEnumerator_TagsLinkedList()
             Assert.True(method.ReturnType.IsValueType);
         }
 
+        [Fact]
+        public void TestParentTraceFlags()
+        {
+            Activity a = new Activity("ParentFlagsA");
+            a.SetIdFormat(ActivityIdFormat.W3C);
+            a.SetParentId(ActivityTraceId.CreateFromString("0123456789abcdef0123456789abcdef".AsSpan()), ActivitySpanId.CreateFromString("0123456789abcdef".AsSpan()), ActivityTraceFlags.Recorded);
+            Assert.Equal("00-0123456789abcdef0123456789abcdef-0123456789abcdef-01", a.ParentId);
+
+            Activity b = new Activity("ParentFlagsB");
+            b.SetIdFormat(ActivityIdFormat.W3C);
+            b.SetParentId(ActivityTraceId.CreateFromString("0123456789abcdef0123456789abcdef".AsSpan()), ActivitySpanId.CreateFromString("0123456789abcdef".AsSpan()), ActivityTraceFlags.None);
+            b.ActivityTraceFlags = ActivityTraceFlags.Recorded; // Setting ActivityTraceFlags shouldn't affect the parent
+            Assert.Equal("00-0123456789abcdef0123456789abcdef-0123456789abcdef-00", b.ParentId);
+        }
```

Comment:
Does this case work? I am suspicious it won't yet
```C#
ActivityTraceId parentContext = new ActivityContext(
    ActivityTraceId.CreateFromString("0123456789abcdef0123456789abcdef".AsSpan()), 
    ActivitySpanId.CreateFromString("0123456789abcdef".AsSpan()), 
    ActivityTraceFlags.Recorded);
    using Activity? c = source.StartActivity(parentContext);
    Assert.Equal("00-0123456789abcdef0123456789abcdef-0123456789abcdef-00", c.ParentId);
```

Reply:
You were right. I have fixed that case too. Thanks for the catch